src/activity_handlers.cpp

```diff
@@ -2687,6 +2687,13 @@ void activity_handlers::mend_item_finish( player_activity *act, player *p )
     if( act->name == "fault_gun_blackpowder" || act->name == "fault_gun_dirt" ) {
         target->set_var( "dirt", 0 );
     }
+
+    //get skill list from mending method, iterate through and give xp
+    const std::map<skill_id, int> skills_trained = method->skills;
+    for( const auto &e : skills_trained ) {
+        p->practice( e.first, 10 );
```

Comment:
```suggestion
        p->practice( e.first, e.second );
```

Reply:
Ah, this is more of what I was looking for:
```suggestion
        p->practice( e.first, 10, static_cast<int>( e.second * 1.25 ) );
```
This follows crafting and limits XP when you're above the level of the skill.